Makefile

```diff
@@ -103,69 +101,84 @@ endif
 help::
 	@echo '  BUILD_MODE: "release", "debug", or "optimized-debug" (Default: "release")'
 
-#
-# address sanitizer
-#
+# ************************************************************************
+# Address sanitizer
+#   - Ctests will be silently skipped with clang builds. (Known issue.)
+#   - Use gcc to build in Asan mode to run unit-tests.
+#   - Tests will run slow in address sanitizer builds.
 ifndef BUILD_ASAN
-BUILD_ASAN=0
+   BUILD_ASAN=0
 endif
 
 ifeq "$(BUILD_ASAN)" "1"
-CFLAGS  += -fsanitize=address
-LDFLAGS += -fsanitize=address
-BUILD_DIR:=$(BUILD_DIR)-asan
-else ifeq "$(BUILD_ASAN)" "0"
-else
-$(error Unknown BUILD_ASAN mode "$(BUILD_ASAN)".  Valid values are "0" or "1". Default is "0")
+   CFLAGS  += -fsanitize=address
+   LDFLAGS += -fsanitize=address
+   BUILD_DIR:=$(BUILD_DIR)-asan
+else ifneq "$(BUILD_ASAN)" "0"
+   $(error Unknown BUILD_ASAN mode "$(BUILD_ASAN)".  Valid values are "0" or "1". Default is "0")
 endif
 
 help::
 	@echo '  BUILD_ASAN={0,1}: Disable/enable address-sanitizer (Default: disabled)'
+	@echo '                    Use gcc to run unit-tests with ASAN-builds.'
 
-#
-# memory sanitizer
-#
+# ************************************************************************
+# Memory sanitizer
+#   - Builds will fail with gcc due to compiler error. Use clang instead.
+#   - Tests will run even slower in memory sanitizer builds.
 ifndef BUILD_MSAN
-BUILD_MSAN=0
+   BUILD_MSAN=0
 endif
 
 ifeq "$(BUILD_MSAN)" "1"
-CFLAGS  += -fsanitize=memory
-LDFLAGS += -fsanitize=memory
-BUILD_DIR:=$(BUILD_DIR)-msan
+   CFLAGS  += -fsanitize=memory
+   LDFLAGS += -fsanitize=memory
+   BUILD_DIR:=$(BUILD_DIR)-msan
 else ifneq "$(BUILD_MSAN)" "0"
-$(error Unknown BUILD_MSAN mode "$(BUILD_MSAN)".  Valid values are "0" or "1". Default is "0")
+   $(error Unknown BUILD_MSAN mode "$(BUILD_MSAN)".  Valid values are "0" or "1". Default is "0")
 endif
 
 help::
 	@echo '  BUILD_MSAN={0,1}: Disable/enable memory-sanitizer (Default: disabled)'
+	@echo '                    Use clang for MSAN-builds.'
 
 #
 # Verbosity
 #
 ifndef BUILD_VERBOSE
-BUILD_VERBOSE=0
+   BUILD_VERBOSE=0
 endif
 
 ifeq "$(BUILD_VERBOSE)" "1"
-COMMAND=
-PROLIX=@echo
-BRIEF=@ >/dev/null echo
-BRIEF_FORMATTED=@ >/dev/null echo
-BRIEF_PARTIAL=@ >/dev/null echo
+   COMMAND=
+   PROLIX=@echo
+   BRIEF=@ >/dev/null echo
+   BRIEF_FORMATTED=@ >/dev/null echo
+   BRIEF_PARTIAL=@ >/dev/null echo
 else ifeq "$(BUILD_VERBOSE)" "0"
-COMMAND=@
-PROLIX=@ >/dev/null echo
-BRIEF=@echo
-BRIEF_FORMATTED=@printf
-BRIEF_PARTIAL=@echo -n
+   COMMAND=@
+   PROLIX=@ >/dev/null echo
+   BRIEF=@echo
+   BRIEF_FORMATTED=@printf
+   BRIEF_PARTIAL=@echo -n
 else
-$(error Unknown BUILD_VERBOSE mode "$(BUILD_VERBOSE)".  Valid values are "0" or "1". Default is "0")
+   $(error Unknown BUILD_VERBOSE mode "$(BUILD_VERBOSE)".  Valid values are "0" or "1". Default is "0")
 endif
 
 help::
+	@echo '  Note: setting BUILD_MODE and other flags may further modify BUILD_DIR'
+	@echo
```

Comment:
Why is this help comment here? It feels like it belongs in the `BUILD_DIR` block.

Reply:
It's here because (seems like) `make help` emits messages as it encounters them in the `Makefile`.

The `help` output will now look like this with the message as you see it on L 172;

```
Environment variables controlling the build:
  BUILD_DIR: Base dir name for build outputs (Default: "build").
    $(BUILD_DIR)/obj: object files
    $(BUILD_DIR)/lib: libraries
    $(BUILD_DIR)/bin: executables
  BUILD_MODE: "release" or "debug" (Default: "release")
  BUILD_ASAN={0,1}: Disable/enable address-sanitizer (Default: disabled)
                    Use gcc to run unit-tests with ASAN-builds.
  BUILD_MSAN={0,1}: Disable/enable memory-sanitizer (Default: disabled)
                    Use clang for MSAN-builds.
  Note: setting BUILD_MODE and other flags may further modify BUILD_DIR

  BUILD_VERBOSE={0,1}: Disable/enable verbose output (Default: disabled)
```

I wanted this "Note: " line to appear -after- all the help messages for other `BUILD_*` env-vars are emitted.